gadopt/stokes_integrators.py

```diff
@@ -205,6 +205,8 @@ class StokesSolver:
       free_surface_theta: Timestepping prefactor for free surface equation, where
                           theta = 0: Forward Euler, theta = 0.5: Crank-Nicolson (default),
                           or theta = 1: Backward Euler
+      nullspace:
+        Dictionary of nullspace options, including transpose and near nullspaces
```

Comment:
I do like that this is explicit now. I'm as yet undecided on the API change, but I guess this kind of technically *reduces* the number of arguments to init, which I think is better.

On this comment, maybe explicitly state the supported nullspaces (`nullspace`, `transpose_nullspace`, `near_nullspace`) and that they can be created with `create_stokes_nullspace`? Or not :shrug:, I don't know the discoverability of this vs. starting with a demo...

Maybe (for later) we could have an abstraction over the interaction `create_stokes_nullspace` and this parameter. Just thinking out loud now!

Reply:
I agree with the need for a more detailed docstring. As an example, [this](https://github.com/g-adopt/g-adopt/blob/aaea809421a3d7cc350888eec9a1047d885763ca/gadopt/stokes_integrators.py#L126-L175) is what I intend to suggest for the coming ABC. Adding another table for null spaces seems a reasonable option.

Regarding `create_stokes_nullspace`, I thought a few times about hiding it from the user, but I never got anywhere I liked. The main issue lies with compressing the arguments that function needs, but I feel like it is not necessarily a great idea.